src/middlewares/user.ts

```diff
@@ -365,155 +356,3 @@ export const checkUserHasSelectedAnOrganizationMiddleware = (
-            await getSelectedOrganizationId(user_id);
-
-          organizationThatNeedsAuthenticationByPeers = userOrganisations.find(
-            ({ id, authentication_by_peers_type }) =>
-              !authentication_by_peers_type && id === selectedOrganizationId,
-          );
-        } else {
-          organizationThatNeedsAuthenticationByPeers = userOrganisations.find(
-            ({ authentication_by_peers_type }) => !authentication_by_peers_type,
-          );
-        }
-
-        if (!isEmpty(organizationThatNeedsAuthenticationByPeers)) {
-          const organization_id = organizationThatNeedsAuthenticationByPeers.id;
-          const internalActiveUsers =
-            await getInternalActiveUsers(organization_id);
-          const otherInternalUsers = internalActiveUsers.filter(
-            ({ email: e }) => e !== email,
-          );
-
-          if (PAIR_AUTHENTICATION_WHITELIST.includes(getEmailDomain(email))) {
-            await markAsWhitelisted({ user_id, organization_id });
-          } else if (usesAGouvFrDomain(email)) {
-            await markAsGouvFrDomain({ user_id, organization_id });
-          } else if (otherInternalUsers.length > 0) {
-            return res.redirect(`/users/choose-sponsor/${organization_id}`);
-          } else {
-            await notifyAllMembers({ user_id, organization_id });
-          }
-        }
-
-        return next();
-      } catch (error) {
-        next(error);
-      }
-    },
-  );
-
-export const checkUserHasBeenGreetedForJoiningOrganizationMiddleware = (
-  req: Request,
-  res: Response,
-  next: NextFunction,
-) =>
-  checkUserHasBeenAuthenticatedByPeersMiddleware(req, res, async (error) => {
-    try {
-      if (error) return next(error);
-
-      const userOrganisations = await getOrganizationsByUserId(
-        getUserFromAuthenticatedSession(req).id,
-      );
-
-      let organizationThatNeedsGreetings;
-      if (req.session.mustReturnOneOrganizationInPayload) {
-        const selectedOrganizationId = await getSelectedOrganizationId(
-          getUserFromAuthenticatedSession(req).id,
-        );
-
-        organizationThatNeedsGreetings = userOrganisations.find(
-          ({ id, has_been_greeted }) =>
-            !has_been_greeted && id === selectedOrganizationId,
-        );
-      } else {
-        organizationThatNeedsGreetings = userOrganisations.find(
-          ({ has_been_greeted }) => !has_been_greeted,
-        );
-      }
-
-      if (!isEmpty(organizationThatNeedsGreetings)) {
-        await greetForJoiningOrganization({
-          user_id: getUserFromAuthenticatedSession(req).id,
-          organization_id: organizationThatNeedsGreetings.id,
-        });
-
-        return res.redirect(
-          `/users/welcome/${organizationThatNeedsGreetings.id}`,
-        );
-      }
-
-      return next();
-    } catch (error) {
-      next(error);
-    }
-  });
-
-// check that user go through all requirements before issuing a session
```

Comment:
keep it!

Reply:
Why do we need to keep the `checkUserHasBeenGreetedForJoiningOrganisationMiddleware` function if it only uses the `checkUserHasBeenAuthenticatedByPeersMiddleware` function? 
Maybe it's a naming problem

Reply:
@rebeccadumazert same here.